integration/obscurogateway/obscurogateway_test.go

```diff
@@ -197,3 +467,165 @@ func waitServerIsReady(serverAddr string) error {
 	}
 	return fmt.Errorf("timed out before server was ready")
 }
+
+func joinObscuroGateway(url string) (string, error) {
+	statusCode, userID, err := fasthttp.Get(nil, fmt.Sprintf("%s/v1/join/", url))
+	if err != nil || statusCode != 200 {
+		return "", fmt.Errorf(fmt.Sprintf("Failed to get userID. Status code: %d, err: %s", statusCode, err))
+	}
+	return string(userID), nil
+}
+
+func registerAccount(url string, userID string, pk *ecdsa.PrivateKey, hexAddress string) ([]byte, error) {
+	payload := prepareRegisterPayload(userID, pk, hexAddress)
+
+	req, err := http.NewRequestWithContext(
+		context.Background(),
+		http.MethodPost,
+		url+"/v1/authenticate/?u="+userID,
+		strings.NewReader(payload),
+	)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json; charset=UTF-8")
+
+	client := &http.Client{}
+	response, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+	return io.ReadAll(response.Body)
+}
+
+func prepareRegisterPayload(userID string, pk *ecdsa.PrivateKey, hexAddress string) string {
+	message := fmt.Sprintf("Register %s for %s", userID, strings.ToLower(hexAddress))
+	prefixedMessage := fmt.Sprintf("\u0019Ethereum Signed Message:\n%d%s", len(message), message)
+	messageHash := crypto.Keccak256([]byte(prefixedMessage))
+	sig, err := crypto.Sign(messageHash, pk)
+	if err != nil {
+		fmt.Printf("Failed to sign message: %v\n", err)
+	}
+	sig[64] += 27
+	signature := "0x" + hex.EncodeToString(sig)
+	payload := fmt.Sprintf("{\"signature\": \"%s\", \"message\": \"%s\"}", signature, message)
+	return payload
+}
+
+func ComputeContractAddress(sender gethcommon.Address, nonce uint64) (gethcommon.Address, error) {
+	// RLP encode the byte array of the sender's address and nonce
+	encoded, err := rlp.EncodeToBytes([]interface{}{sender, nonce})
+	if err != nil {
+		return gethcommon.Address{}, err
+	}
+	// Compute the Keccak-256 hash of the RLP encoded byte array
+	hash := crypto.Keccak256(encoded)
+
+	// The contract address is the last 20 bytes of this hash
+	return gethcommon.BytesToAddress(hash[len(hash)-20:]), nil
+}
+
+func TransferETHToAddress(client *ethclient.Client, wallet wallet.Wallet, toAddress gethcommon.Address, amount int64) error {
```

Comment:
What if the transactions were awaited, and confirmed sucessfull before returning ?

Reply:
Added awaiting for transactions